tests/dialects/builtin_test.py

```diff
@@ -0,0 +1,30 @@
+from xdsl.dialects.builtin import *
+
+
+def test_DenseIntOrFPElementsAttr():
+    check1 = DenseIntOrFPElementsAttr.tensor_from_list([4, 5], f32)
+
+    value1 = check1.data.data[0].value.data
+    value2 = check1.data.data[1].value.data
+
+    # Ensure type conversion happened properly during attribute construction.
+    res1 = type(value1) == float and value1 == 4.0 and type(
+        value2) == float and value2 == 5.0
+    assert (res1)
```

Comment:
AFAIU, you are not using `res1` anywhere else. Therefore, I think it should be replaced by its definition. Additionally, I think it is easier if each individual part of this four-way conjunction is its own assertion. Then one immediately sees which condition was wrong.

Reply:
Same for res2 and the other assertion.